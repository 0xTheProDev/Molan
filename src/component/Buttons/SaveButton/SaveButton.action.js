// @flow
import { saveAs } from 'save-as';
import { MIME_TYPES, FILE_EXTENSIONS } from 'constants/templates';

export function onSaveButtonClick(language: string, content: string) {
  return function onClickHandler(event: MouseEvent): void {
    const MimeType = MIME_TYPES[language] || MIME_TYPES['text'];
    const Extension = FILE_EXTENSIONS[language] || FILE_EXTENSIONS['text'];

    const blob = new Blob([ content ], { type: MimeType });
    const FileName = `molan${Extension}`;

    saveAs(blob, FileName);
  }
}
