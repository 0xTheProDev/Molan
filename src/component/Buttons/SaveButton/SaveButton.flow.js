// @flow
export type SaveButtonProps = {
  language: string,
  content: string,
  SaveButtonText: string,
};

export type SaveButtonInternalProps = {
  SaveButtonText: string,
  onSaveButtonClick: (event: MouseEvent) => void,
};
