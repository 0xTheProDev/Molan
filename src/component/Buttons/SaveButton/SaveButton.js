// @flow
import React from 'react';
import { Button, Icon } from 'semantic-ui-react';
import { noop } from 'util/common';

import { SaveButtonInternalProps } from './SaveButton.flow';

export function SaveButton({ SaveButtonText, onSaveButtonClick }: SaveButtonInternalProps) {
  return (
    <Button
      animated="fade"
      floated="right"
      title={SaveButtonText}
      onClick={onSaveButtonClick}
    >
      <Button.Content hidden>{SaveButtonText}</Button.Content>
      <Button.Content visible>
        <Icon name="save" />
      </Button.Content>
    </Button>
  );
}

SaveButton.defaultProps = {
  SaveButtonText: 'Save',
  onSaveButtonClick: noop,
};
