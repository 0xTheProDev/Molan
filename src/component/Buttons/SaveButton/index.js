// @flow
import React from 'react';

import { SaveButtonProps } from './SaveButton.flow';
import { onSaveButtonClick } from './SaveButton.action';
import { SaveButton } from './SaveButton';

export default function SaveButtonFactory(props: SaveButtonProps) {
  const {
    language,
    content,
    SaveButtonText,
  } = props;

  const onClickHandler = onSaveButtonClick(props.language, props.content);

  return (
    <SaveButton
      SaveButtonText={SaveButtonText}
      onSaveButtonClick={onClickHandler}
    />
  );
}

export { SaveButtonProps, SaveButtonFactory as SaveButton };
