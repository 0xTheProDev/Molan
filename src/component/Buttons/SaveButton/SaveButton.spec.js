import React from 'react';
import { Button } from 'semantic-ui-react';

import { SaveButton } from './SaveButton';

describe('Component::Buttons:SaveButton::SaveButton', () => {
  let
    onSaveButtonClickStub,
    minProps;

  beforeAll(() => {
    onSaveButtonClickStub = jest.fn();

    minProps = {
      SaveButtonText: 'Save',
      onSaveButtonClick: onSaveButtonClickStub,
    };
  });

  it('should render with base structure', () => {
    const wrapper = shallow(<SaveButton {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke callback handler when button clicked', () => {
    const wrapper = shallow(<SaveButton {...minProps} />);

    wrapper.find(Button).simulate('click');

    expect(onSaveButtonClickStub).toHaveBeenCalled();
  });
});
