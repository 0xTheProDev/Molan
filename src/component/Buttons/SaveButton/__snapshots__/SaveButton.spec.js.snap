// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::Buttons:SaveButton::SaveButton should render with base structure 1`] = `
<Button
  animated="fade"
  as="button"
  floated="right"
  onClick={[MockFunction]}
  title="Save"
>
  <ButtonContent
    hidden={true}
  >
    Save
  </ButtonContent>
  <ButtonContent
    visible={true}
  >
    <Icon
      as="i"
      name="save"
    />
  </ButtonContent>
</Button>
`;
